src/traits.rs

```diff
@@ -871,6 +939,22 @@ macro_rules! prim_int_impl {
                 <$T>::rotate_right(self, n)
             }
 
+            fn arithmetic_shl(self, n: u32) -> Self {
+                ((self as $S) << n) as $T
```

Comment:
If we don't do "fake" sign extension, then there is no point to these new functions at all -- just use `Shr`.  Rather, I am treating it something like x86 `shar` which extends the most significant bit, as if it were signed, even though raw registers are not inherently signed/unsigned necessarily.

So I did document this -- "arithmetic" here is bitwise equivalent to signed shift, and "logical" is bitwise equivalent to unsigned shift.  The example explicitly shows the "sign"-extension even on u64, and doctest will verify this.